rules/template-indent.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+const os = require('os');
+const stripIndent = require('strip-indent');
+const indentString = require('indent-string');
+const {replaceTemplateElement} = require('./fix/index.js');
+const {matches} = require('./selectors/index.js');
+
+const MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE = 'template-indent';
+const messages = {
+	[MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE]: 'Templates should be properly indented.',
+};
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	const options = {
+		tags: ['outdent', 'dedent', 'gql', 'sql', 'html', 'styled'],
+		functions: ['dedent', 'stripIndent'],
+		selectors: [],
+		comments: ['HTML', 'indent'],
+		...context.options[0],
+	};
+
+	options.comments = options.comments.map(comment => comment.toLowerCase());
+
+	const selectors = [
+		...options.tags.map(tag => `TaggedTemplateExpression[tag.name="${tag}"] > .quasi`),
+		...options.functions.map(fn => `CallExpression[callee.name="${fn}"] > .arguments:first-child`),
+		...options.selectors,
+	];
+
+	const handled = new Set();
+
+	/** @param {import('@babel/core').types.TemplateLiteral} node */
+	const templateLiteralHandler = node => {
+		if (handled.has(node)) {
+			return;
+		}
+
+		handled.add(node);
+
+		if (node.type !== 'TemplateLiteral') {
+			return;
+		}
+
+		const delimiter = '__PLACEHOLDER__' + Math.random();
+		const joined = node.quasis.map(q => q.value.raw).join(delimiter);
+
+		if (!joined.includes('\n')) {
+			return;
+		}
+
+		const startLine = context.getSourceCode().lines[node.loc.start.line - 1];
+		const marginMatch = startLine.match(/^(\s*)\S/);
+		const parentMargin = marginMatch ? marginMatch[1] : '';
+
+		let indent;
+		if (typeof options.indent === 'string') {
+			indent = options.indent;
+		} else if (typeof options.indent === 'number') {
+			indent = ' '.repeat(options.indent);
+		} else {
+			const tabs = parentMargin.startsWith('\t');
+			indent = tabs ? '\t' : '  ';
+		}
+
+		const dedented = stripIndent(joined);
+		const fixed
+			= os.EOL
+			+ indentString(dedented.trim(), 1, {indent: parentMargin + indent})
+			+ os.EOL
+			+ parentMargin;
+
+		if (fixed === joined) {
+			return;
+		}
+
+		context.report({
+			node,
+			messageId: MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE,
+			fix: fixer => fixed
+				.split(delimiter)
+				.map((replacement, index) => replaceTemplateElement(fixer, node.quasis[index], replacement)),
+		});
+	};
+
+	const matchers = {
+		[matches(selectors)]: templateLiteralHandler,
+	};
+
+	if (options.comments.length > 0 && !selectors.includes('TemplateLiteral')) {
```

Comment:
This is not good enough, because `TemplateLiteral` can be written in other way, `[type="TemplateLiteral"]`, `[type='TemplateLiteral']`, `[type=/^TemplateLiteral$/]` etc, can you try the solution I proposed https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1478#discussion_r687536214?

Reply:
This check is just to make sure we don't override the existing handler, I was trying to avoid adding a handler for every template literal when not necessary. But yes I can switch to your suggestion, since there are comments in the default config we will need to look at every template literal almost all the time.